classes/migration/upgrade/v3_4_0/PreflightCheckMigration.php

```diff
@@ -35,163 +34,249 @@ protected function getContextSettingsTable(): string
-            foreach ($orphanedIds as $userId) {
-                DB::table('subscriptions')->where('user_id', '=', $userId)->delete();
-            }
+        $this->addTableProcessor('publication_galley_settings', function (): int {
+            $affectedRows = 0;
+            // Depends directly on ~1 entities: galley_id->publication_galleys.galley_id
+            // Custom field (not found in at least one of the softwares)
+            $affectedRows += $this->deleteRequiredReference('publication_galley_settings', 'galley_id', 'publication_galleys', 'galley_id');
+            return $affectedRows;
+        });
 
-            // Clean orphaned subscriptions entries by type_id
-            $orphanedIds = DB::table('subscriptions AS s')->leftJoin('subscription_types AS st', 'st.type_id', '=', 's.type_id')->whereNull('st.type_id')->distinct()->pluck('s.type_id');
-            foreach ($orphanedIds as $typeId) {
-                DB::table('subscriptions')->where('type_id', '=', $typeId)->delete();
-            }
+        $this->addTableProcessor('section_settings', function (): int {
+            $affectedRows = 0;
+            // Depends directly on ~1 entities: section_id->sections.section_id
+            // Custom field (not found in at least one of the softwares)
+            $affectedRows += $this->deleteRequiredReference('section_settings', 'section_id', 'sections', 'section_id');
+            return $affectedRows;
+        });
 
-            // Clean orphaned institutional_subscriptions entries by subscription_id
-            $orphanedIds = DB::table('institutional_subscriptions AS i_s')->leftJoin('subscriptions AS s', 'i_s.subscription_id', '=', 's.subscription_id')->whereNull('s.subscription_id')->distinct()->pluck('i_s.subscription_id');
-            foreach ($orphanedIds as $subscriptionId) {
-                DB::table('institutional_subscriptions')->where('subscription_id', '=', $subscriptionId)->delete();
-            }
+        $this->addTableProcessor('subscription_type_settings', function (): int {
+            $affectedRows = 0;
+            // Depends directly on ~1 entities: type_id->subscription_types.type_id
+            // Custom field (not found in at least one of the softwares)
+            $affectedRows += $this->deleteRequiredReference('subscription_type_settings', 'type_id', 'subscription_types', 'type_id');
+            return $affectedRows;
+        });
 
-            // Clean orphaned completed_payments entries by context_id
-            $orphanedIds = DB::table('completed_payments AS cp')->leftJoin('journals AS j', 'j.journal_id', '=', 'cp.context_id')->whereNull('j.journal_id')->distinct()->pluck('cp.context_id');
-            foreach ($orphanedIds as $journalId) {
-                DB::table('completed_payments')->where('context_id', '=', $journalId)->delete();
+        // Support for the issueId setting
+        $this->addTableProcessor('publication_settings', function (): int {
```

Comment:
Here I decided to delete the `issueId` when it's invalid. This is a fresh update, I didn't check the outcome yet,